Source/Source/CompVendingMachine.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using RimWorld;
+using UnityEngine;
+using Verse;
+using FoodUtility = Hospitality.Utilities.FoodUtility;
+
+namespace Hospitality
+{
+    public class CompVendingMachine : ThingComp, IThingHolder
+    {
+        private bool isActive = false;
+
+        private int basePrice = 10;
+
+        private ThingOwner<Thing> silverContainer;
+
+        public Building_NutrientPasteDispenser Dispenser => (Building_NutrientPasteDispenser) parent;
+
+        public int CurrentPrice
+        {
+            get => basePrice;
+            internal set => basePrice = Mathf.Clamp(value, 0, int.MaxValue);
+        }
+
+        public bool IsFree => CurrentPrice == 0;
+        public string TotalSold => $"${silverContainer.TotalStackCount}";
```

Comment:
This should use the convert to currency function of RimWorld

Reply:
Resolved using the .ToStringMoney() extension.